deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md

```diff
@@ -0,0 +1,190 @@
+	```
+
+	Sample `info` call:
+
+	[https://api.polkaholic.io/info/shibuya/0x396418ff533172de8407004f53051b5409f6892564ddeba12f75cc855cb16fe0](https://api.polkaholic.io/info/shibuya/0x396418ff533172de8407004f53051b5409f6892564ddeba12f75cc855cb16fe0)
+
+
+### ChainIDE-Polkaholic Verification Build Process: from Milestone 1 to Milestone 2
+
+ChainIDE uses a containerized build process, where tiny variations in configuration may result in different codehashes.  This is acceptable for the above security model where ChainIDE is currently assumed to be a trusted IDE.   However, to standardize WASM Contract compilation such that multiple parties (in particular Colorful Notion and ChainIDE as well as others) can use identical build processes to derive the exact same code hash, in July 2023 Parity released the following  "cargo contract build --verifiable" functionality:
+**[https://github.com/paritytech/cargo-contract/releases/tag/v4.0.0-alpha](https://github.com/paritytech/cargo-contract/releases/tag/v4.0.0-alpha)**
+The official version is expected to be launched by the end of the year.  Both teams researched the usability of this (ChainIDE to incorporate into its build process, Polkaholic to replicate and arrive).   See issue [here](TODO).
+
+We propose to collaborate _with Parity's assistance_ in Milestone 2 using this verifiable build process, wherein:
+* ChainIDE will use Parity's "cargo contract build --verifiable" process, and use the same `/verify` API endpoint
+* Polkaholic will verify the uploaded code using its the identical "--verifiable" process
+
+
+
+### Polkaholic.io API for WASM Contracts
+
+The [Polkaholic.io API](https://docs.polkaholic.io/#introduction) has been extended to include the following WASM Contract APIs at the same REST API Endpoint of [https://api.polkaholic.io](https://api.polkaholic.io) to enable anyone to access data about a network and contract address with simple `{network}` and `{contractAddress}` parameters.
+
+|Function|REST API Format|Example|
+|---|---|--|
+|Retrieve indexed code hashes by code hash | https://api.polkaholic.io/wasmcode/`{network}` | https://api.polkaholic.io/wasmcode/shiden |
+|Retrieve contract addresses | https://api.polkaholic.io/wasmcontracts/`{network}` | https://api.polkaholic.io/wasmcontracts/shibuya |
+|  Retrieve contract calls of a contract address |  https://api.polkaholic.io/wasmcontract/`{network}`/`{contractAddress}`  | https://api.polkaholic.io/wasmcontract/shiden/XRcGZzdH841dHySiM62BfJDbQfXXYex3U7LYooRJt8EgJeX |
```

Comment:
This example URL is throwing a 404 error

```Cannot GET /wasmcontract/shiden/XRcGZzdH841dHySiM62BfJDbQfXXYex3U7LYooRJt8EgJeX```

Reply:
Ya this is still an issue for me

Reply:
Apologies, this is fixed -- url is actually 
 
 https://api.polkaholic.io/wasmcontract/shibuya/XRcGZzdH841dHySiM62BfJDbQfXXYex3U7LYooRJt8EgJeX